Test mies nwb data

sgratiy opened the pull request:
Tests for EphysNWBData and MiesNWBData. Addressing #363

NileGraddis commented on ipfx/dataset/ephys_nwb_data.py:
```diff
@@ -51,23 +51,17 @@ class EphysNWBData(EphysDataInterface):
 
-        super().init(ontology=ontology, validate_stim=validate_stim)
+        super().__init__(ontology=ontology)
+        load_into_memory = False # not working
```
is this a todo?

sgratiy replied:
@gnayuy has fixed this on his branch

NileGraddis commented on ipfx/dataset/ephys_data_interface.py:
```diff
@@ -189,15 +162,13 @@ def get_stimulus_unit(self, sweep_number: int) -> str:
 
     @abc.abstractmethod
-    def get_real_sweep_number(self, sweep_name:str, assumed_sweep_number: Optional[int]=None) -> int:
+    def get_sweep_number(self, sweep_name:str) -> int:
```
can this be deleted?

sgratiy replied:
I believe so. The logic was to try getting the "real" (explicitly assigned) sweep number. But for old nwb1 files it was missing, in which case resort to inferring sweep number from the dataset name. My understanding that for nwb2 "real" sweep number is always exists.

NileGraddis replied:
ok, would you mind deleting then?

sgratiy replied:
I thought we were talking about `get_real_sweep_number`, but you want me to delete `get_sweep_number`. got it.

NileGraddis commented on ipfx/dataset/ephys_nwb_data.py:
```diff
@@ -192,37 +176,30 @@ def get_session_start_time(self):
 
-            return unit_str
+        for s in series:
+            if isinstance(s,(CurrentClampStimulusSeries,VoltageClampStimulusSeries)):
```
is there a common base type?

sgratiy replied:
There is (PatchClampSeries), but it is also the parent for the CurrentClampSeries, VoltageClampSeries used by the response series.

NileGraddis commented on ipfx/dataset/ephys_nwb_data.py:
```diff
@@ -192,37 +176,30 @@ def get_session_start_time(self):
+        if len(stimulus) == 1:
+            return stimulus[0].unit
+        elif len(stimulus)>1:
+            raise ValueError("Found multiple stimulus series for sweep number {sweep_number}")
```
would be good if the error message identified them somehow

NileGraddis commented on ipfx/stimulus.py:
```diff
@@ -26,7 +26,7 @@ def has_tag(self, tag, tag_type=None):
 class StimulusOntology(object):
 
-    DEFAULT_STIMULUS_ONTOLOGY_FILE = os.path.join(os.path.dirname(__file__), 'defaults/stimulus_ontology.json')
+    DEFAULT_STIMULUS_ONTOLOGY_FILE = os.path.join(os.path.dirname(__file__), './defaults/stimulus_ontology.json')
```
probably don't want to hardcode these forward slashes

NileGraddis replied:
what about the one between defaults and stimulus ontology?

sgratiy replied:
I am not sure how to point to the json otherwise

sgratiy commented:
@NileGraddis could you please re-review my PR. I wiped out all the sweep_map junk and added get_clamp_mode and a helper function to access series based on sweep_number `EphysNWBData._get_series`.  Since I rebased I fully re-written the history.

NileGraddis commented on ipfx/dataset/create.py:
```diff
@@ -73,12 +77,16 @@ def create_ephys_data_set(nwb_file: str,
+            nwb_data = HBGNWB2Data(nwb_file, ontology)
 
     else:
         raise ValueError("Unsupported or unknown NWB major" +
```
use formatting here rather than +

NileGraddis replied:
(I realize this is not your change, but in the interests of good stewardship would be nice, not a requirement to merge).

NileGraddis commented on ipfx/dataset/ephys_data_interface.py:
```diff
@@ -130,37 +131,6 @@ def get_stimulus_unit(self, sweep_number: int) -> str:
 
         raise NotImplementedError
 
-    @abc.abstractmethod
```
nice!

NileGraddis commented on ipfx/dataset/ephys_nwb_data.py:
```diff
@@ -51,30 +51,58 @@ class EphysNWBData(EphysDataInterface):
-                    "stimulus_scale_factor",
-                    "bridge_balance",
-                    "stimulus_scale_factor"]
+    STIMULUS = (VoltageClampStimulusSeries, CurrentClampStimulusSeries)
```
good stuff

NileGraddis commented on ipfx/dataset/ephys_nwb_data.py:
```diff
@@ -51,30 +51,58 @@ class EphysNWBData(EphysDataInterface):
 
-        super().init(ontology=ontology, validate_stim=validate_stim)
+        super().__init__(ontology=ontology)
+        load_into_memory = False # not working
```
If this is not working, delete it and raise a ticket.
